cryptoki/src/mechanism/vendor_defined.rs

```diff
@@ -0,0 +1,37 @@
+// Copyright 2024 Contributors to the Parsec project.
+// SPDX-License-Identifier: Apache-2.0
+//! Mechanism types are defined with the objects and mechanism descriptions that use them.
+//! Vendor defined values for this type may also be specified.
+//! See: <https://docs.oasis-open.org/pkcs11/pkcs11-base/v3.0/os/pkcs11-base-v3.0-os.html#_Toc29976545>
```

Comment:
Are vendor mechanisms only supported in v3 of PKCS #11?

Reply:
It's also supported in v2.4. I put v3.0 link here since we are using v3.0 headers now.